### ISSUE TITLE
Mention text and formatting desynchronize

### ISSUE THREAD

**Issue by roancoetzee** [signavio/react-mentions#388]
**Steps to reproduce:**

1. Go to https://react-mentions.now.sh/
2. Add some lorem ipsum text to any of the multiline textareas (Multiple Trigger Patterns, Scrollable Container etc)
3. Add a few mentions after the text

The next step is dependent on how much text you added and where the mention is located within that text 

4. Change the browser resolution  until the you see the desync happen.

**Expected behaviour:**
- Mention text and format should always be in synch
![Screenshot 2020-03-20 at 09 24 10](https://user-images.githubusercontent.com/2031707/77150833-a84ff700-6a8c-11ea-88f7-e745f3024158.png)

**Observed behaviour:**
- At certain textarea/resolution widths the mention and the format for that mention desync. See attached image.
![Screenshot 2020-03-20 at 09 23 59](https://user-images.githubusercontent.com/2031707/77150822-a128e900-6a8c-11ea-9aea-e8ae7f62a58a.png)

**Workaround:**
N/A

**Comment by khamiltonuk** [comment]
@roancoetzee, thanks for the issue, I was wondering  how do change the browser resolution? I am trying to replicate the issue

- do you mean resizing the browser?
- changing the browser orientation by rotating a device?
- dragging the browser window between two monitors or different  resolutions

**Comment by roancoetzee** [comment]
@khamiltonuk Yes resizing the browser. I'll update the issue.

**Comment by jsheffers** [comment]
Agreed. Here is a screenshot from the examples page:
<img width="223" alt="Screen Shot 2020-04-06 at 12 14 54 PM" src="https://user-images.githubusercontent.com/656483/78580426-5fc06980-7800-11ea-93a4-03edf6e84c24.png">

Also, in my own app, I'm having shifting issues as well.  

<img width="698" alt="Screen Shot 2020-04-06 at 12 16 26 PM" src="https://user-images.githubusercontent.com/656483/78580513-7d8dce80-7800-11ea-8019-e44164926c2b.png">

Any ideas?

**Comment by yurikelvin** [comment]
I had the same issue but in a different scenario. When you add some text like lorem ipsum text and in the middle of the text you put a mention and delete/add a few characters before the mention until the break line the mention text and formatting desynchronize. In my case the shifting issues it's based on pipe of the keyboard (put pipe right across the break line and add a mention too). 

![Screenshot from 2020-04-27 17-46-32](https://user-images.githubusercontent.com/24192482/80419466-95acb700-88af-11ea-8fd7-a3a0a16ce376.png)

![Screenshot from 2020-04-27 17-49-40](https://user-images.githubusercontent.com/24192482/80419472-99403e00-88af-11ea-8cb4-d641c295c536.png)

Resolution tested: 1920x1080

Sorry about my English, hope that I helped.

**Comment by theo-m** [comment]
Maybe I can offer help: we've struggled on this too with an overlay like the one in [this file](https://github.com/signavio/react-mentions/blob/master/demo/src/examples/CssModules.js) to have the mentions show up like links. 
Our solution was to set 
```css
text-rendering: geometricPrecision;
```
in the text-area

**Comment by AlexBHarley** [comment]
I'm also having this issue. The more newlines I add the further the highlighting is out of sync

<img width="439" alt="Screen Shot 2020-05-24 at 22 00 11" src="https://user-images.githubusercontent.com/11168524/82763736-0d67f600-9e0a-11ea-9a5c-0a2a7bcb7c54.png">

**Comment by basyusuf** [comment]
Anybody find a solution?

**Comment by Seoudi** [comment]
Yes  as attached below the styling for the mentions__highlighter and mentions__input must be identical I think this should solve the problem and you can reset default styling in them maybe it affects one of them only
![image](https://user-images.githubusercontent.com/17534974/87236310-0a3fbe00-c3e8-11ea-851e-940a360e7b02.png)
for more details check this article I created it handles everything from the begining https://medium.com/@aly.essam.seoudi/simple-react-guide-to-implement-chat-comment-input-component-react-mentions-emoji-mart-a48a4d8c4fad

**Comment by Seoudi** [comment]
Yes  as attached below the styling for the mentions__highlighter and mentions__input must be identical I think this should solve the problem and you can reset default styling in them maybe it affects one of them only
![image](https://user-images.githubusercontent.com/17534974/87236310-0a3fbe00-c3e8-11ea-851e-940a360e7b02.png)
for more details check this article I created it handles everything from the begining https://medium.com/@aly.essam.seoudi/simple-react-guide-to-implement-chat-comment-input-component-react-mentions-emoji-mart-a48a4d8c4fad

**Comment by vinrei** [comment]
Anyone found a solution to this one? It is causing issues for me. Our text input has a relatively a narrow width which seems to make it worse as the text is spread over more lines.

**Comment by omeedrabani** [comment]
I tried a lot of different things and noticed that the amount of shifting is very different depending on what font you're using.

Here's the font in https://github.com/signavio/react-mentions/blob/master/demo/src/examples/example.module.css, which I noticed had zero issues with shifting for me (I only tested with a small amount of text).
```css
.mentions--multiLine .mentions__control {
  font-family: monospace;
  font-size: 14pt;
}
```

**Workaround for me**: I was able to get it working for my font by setting `font-weight` for the highlighter css.

I'm using `font-family: "Avenir"` and `font-size: 14px`. 

Depending on what font you're using though, this may or may not work. Good luck! Hopefully this helps 🤞 

```css
.mentions--multiLine .mentions__highlighter {
  font-weight: 400;
}
```

**Comment by omeedrabani** [comment]
Follow up here, depending on what your `word-break` is set to, I also found that if I had `overflow-y: scroll;` set on on `multiLine`, I needed to make sure it was also set on the `highlighter`, as that was causing word wrapping to be problematic in some cases and also causing the overlayed highlighter to be misaligned.

**Comment by MRamanenkau** [comment]
How does it work? Well, it has two layers of text - the visible and the transparent one with the visible highlight bar. These texts and the linebreaks should match so the bar lies under the mention text. But it's not matching because the underlying transparent text has an extra invisible 1px border.
The solution is to find `highlighter` element and disable this border. 
```
&__highlighter {
    border: 0 !important;
  }
```
